tests/test_web_response.py

```diff
@@ -1171,3 +1173,36 @@ def test_text_is_json_encoded(self) -> None:
     def test_content_type_is_overrideable(self) -> None:
         resp = json_response({"foo": 42}, content_type="application/vnd.json+api")
         assert "application/vnd.json+api" == resp.content_type
+
+
+@pytest.mark.skipif(
+    not bool(os.environ.get("PYTHONASYNCIODEBUG")),
+    reason="Only run the test with debug mode",
+)
+async def test_warn_large_cookie(buf: Any, writer: Any) -> None:
+    resp = Response()
+    resp.set_cookie("foo", "8" * 4070, max_age=2600)  # No warning
+    req = make_request("GET", "/", writer=writer)
+
+    await resp.prepare(req)
+    await resp.write_eof()
+
+    cookie = re.search(b"Set-Cookie: (.*?)\r\n", buf).group(1)
+    assert len(cookie) == 4096
+    buf[:] = b""
+
+    resp = Response()
+
+    with pytest.warns(
+        UserWarning,
+        match="The size of is too large, it might get ignored by the client.",
+    ) as record:
+        resp.set_cookie("foo", "8" * 4071, max_age=2600)
+    req = make_request("GET", "/", writer=writer)
+
+    await resp.prepare(req)
+    await resp.write_eof()
+
+    cookie = re.search(b"Set-Cookie: (.*?)\r\n", buf).group(1)
+    assert len(cookie) == 4097
```

Comment:
Could this be something like
```suggestion
    assert re.match(b"Set-Cookie: [^\s]{4097}\r\n", buf)
```
?

Reply:
It could, though I wonder if that's being a little too smart. I feel like the original code is clearer.

Reply:
I think what @webknjaz suggests is more clear.